app/uk/gov/hmrc/eoricommoncomponent/frontend/services/Save4LaterService.scala

```diff
@@ -69,4 +72,17 @@ class Save4LaterService @Inject() (save4LaterConnector: Save4LaterConnector) {
       .get[EmailStatus](internalId.id, emailKey)
   }
 
+  def fetchProcessingService(
+    groupId: GroupId
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Service]] = {
+    logger.info("fetching Processing service from mongo")
+    save4LaterConnector
+      .get[CacheIds](groupId.id, CachedData.groupIdKey)
+      .map {
+        case Some(cacheIds) => cacheIds.serviceCode.flatMap(Service.withName)
+        case _              => throw new IllegalStateException("No cacheIds stored for this group")
```

Comment:
Can this be thrown when user is processing CDS enrolment from Get access to cds service?

Should we handle this as displaying that user is during enrolment to "different service" (without mentioning to what service user has processing enrolment)?